pkg/util/azureclient/applens/applens_client.go

```diff
@@ -0,0 +1,138 @@
+package applens
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+// AppLens Client created from CosmosDB Client
+// (https://github.com/Azure/azure-sdk-for-go/blob/3f7acd20691214ef2cb1f0132f82115f1df01a8c/sdk/data/azcosmos/cosmos_client.go)
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+)
+
+// AppLens client is used to interact with the Azure AppLens service.
+type Client struct {
+	endpoint string
+	pipeline runtime.Pipeline
+}
+
+// Endpoint used to create the client.
+func (c *Client) Endpoint() string {
+	return c.endpoint
+}
+
+// NewClient creates a new instance of AppLens client with Azure AD access token authentication. It uses the default pipeline configuration.
+// endpoint - The applens service endpoint to use.
+// cred - The credential used to authenticate with the applens service.
+// options - Optional AppLens client options.  Pass nil to accept default values.
+func NewClient(endpoint, scope string, cred azcore.TokenCredential, o *ClientOptions) (*Client, error) {
+	return &Client{endpoint: endpoint, pipeline: newPipeline([]policy.Policy{runtime.NewBearerTokenPolicy(cred, []string{fmt.Sprintf("%s/.default", scope)}, nil)}, o)}, nil
+}
+
+func newPipeline(authPolicy []policy.Policy, options *ClientOptions) runtime.Pipeline {
+	if options == nil {
+		options = NewClientOptions()
+	}
+
+	return runtime.NewPipeline("applens", serviceLibVersion,
+		runtime.PipelineOptions{
+			PerCall:  []policy.Policy{},
+			PerRetry: authPolicy,
+		},
+		&options.ClientOptions)
+}
+
+// ListDetectors obtains the list of detectors for a service from AppLens.
+// ctx - The context for the request.
+// o - Options for Read operation.
+func (c *Client) ListDetectors(
+	ctx context.Context,
+	o *ListDetectorsOptions) (*http.Response, error) {
+	if o == nil {
+		o = &ListDetectorsOptions{}
+	}
+
+	azResponse, err := c.sendPostRequest(
+		ctx,
+		o,
+		nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return azResponse, nil
+}
+
+// GetDetector obtains detector information from AppLens.
+// ctx - The context for the request.
+// o - Options for Read operation.
+func (c *Client) GetDetector(
+	ctx context.Context,
+	o *GetDetectorOptions) (*http.Response, error) {
+	if o == nil {
+		o = &GetDetectorOptions{}
+	}
+
+	azResponse, err := c.sendPostRequest(
+		ctx,
+		o,
```

Comment:
`o` is used for `options` all over. Can we rename this to `options`?

Reply:
oh I see, thanks for sharing that wiki. I can push aside my want for more descriptive local variables in favor of that recommendation.